src/js/components/Calendar/Calendar.js

```diff
@@ -182,7 +193,38 @@ const Calendar = forwardRef(
 
     // set date when caller changes it, allows us to change it internally too
     const [date, setDate] = useState(dateProp);
-    useEffect(() => setDate(normalizeForTimezone(dateProp)), [dateProp]);
+    useEffect(() => {
+      if (typeof dateProp === 'string') {
+        setDate(normalizeForTimezone(dateProp));
+      } else if (Array.isArray(dateProp)) {
+        if (Array.isArray(dateProp[0])) {
+          let from;
+          let to;
+          [from, to] = dateProp[0].map(day =>
+            day ? new Date(day) : undefined,
+          );
+          if (from) from = normalizeForTimezone(from, dateProp[0][0]);
+          if (to) to = normalizeForTimezone(to, dateProp[0][0]);
+
+          setDate([[from, to]]);
+        } else {
+          const dateArray = [];
```

Comment:
Since this section is mirroring what is happening in the `dates` useEffect below, is there anyway to extract the functionality into a reusable function that receives the prop (dateProp or datesProp) and the array (dateArray or datesArray)?

Reply:
Yes, sure I give it a try.